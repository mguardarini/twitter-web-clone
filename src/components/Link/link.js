import React from 'react';
import Link from '@material-ui/core/Link';
import useStyles from './style';

export default function CustomLink(props) {
    const classes = useStyles();
    return (
        <Link
            className={[classes.link,props.className]}
            component="button"
            variant="body2"
            onClick={() => {
                alert("I'm footer.");
            }}
            >
              {props.text}
        </Link>
    )
}   
