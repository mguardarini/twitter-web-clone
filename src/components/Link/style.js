import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
    link:{
        fontSize: '12px',
        paddingLeft:'4px',
        paddingRight:'12px',
        marginTop:'5px',
        color:'#aab8c2',
        textAlign:'center',
        fontFamily:'Segoe UI'
    },
  }));

export default useStyles;