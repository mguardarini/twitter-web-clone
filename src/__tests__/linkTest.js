import React from 'react'
import { createMemoryHistory } from "history";
import {render, fireEvent} from '@testing-library/react'
import Link from '../../src/components/Link/link'
 
describe('<Link /> spec', () => {
    it('Renders the component Link', () => {
      const container = render(<Link text='Test' />)
      expect(container.firstChild).toMatchSnapshot()
      expect(container.findByLabelText("Test"))
     })

     it('Should click in Link component', () => {
        const history = createMemoryHistory({ initialEntries: ["/"] });
        const {getByText} = render(<Link text='Home' path='/'/>);    
        const link = getByText('Home');
        fireEvent.click(link);
        expect(history.location.pathname).toBe("/");
    })
});