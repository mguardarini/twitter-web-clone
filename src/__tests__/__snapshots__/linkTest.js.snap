// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> spec Renders the component Link 1`] = `undefined`;

exports[`<Link /> spec Should click in Link component 1`] = `
<button
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-38 MuiLink-button MuiTypography-body2 MuiTypography-colorPrimary"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
>
  Test
</button>
`;
